File: pyg_parent/interface/src/main/java/cn/itcast/core/service/BrandService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.entity.PageResult;
import cn.itcast.core.pojo.good.Brand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    public List<Brand> findAll();

    public PageResult findPage(Brand brand ,Integer page, Integer rows);

    public void add(Brand brand);

    public Brand findOne(Long id);

    public void update(Brand brand);

    public void delete(Long[] ids);

    public List<Map> selectOptionList();

    PageResult search(Integer page,Integer rows,Brand brand, String sellerId);

    void sellerAdd(String sellerId,Brand brand);

    boolean checkBrand(Brand brand);

    Brand findById(String sellerId, Long id);

    void updateSellerBrand(String sellerId, Brand brand);

    PageResult ManagerSearch(Integer page, Integer rows, Brand brand, String sellerId);

    void updateStatus(Long[] ids, String status);
}

File: pyg_parent/web_manager/src/main/webapp/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){
	this.findAll = function(){
		return $http.get("../brand/findAll.do");
	}
	
	this.findByPage = function(page,rows){
		return $http.get("../brand/findByPage.do?page="+page+"&rows="+rows);
	}
	
	this.save = function(entity){
		return $http.post("../brand/add.do",entity);
	}
	
	this.update=function(entity){
		return $http.post("../brand/update.do",entity);
	}
	
	this.findById=function(id){
		return $http.get("../brand/findOne.do?id="+id);
	}
	
	this.dele = function(ids){
		return $http.get("../brand/delete.do?ids="+ids);
	}
	
	this.search = function(page,rows,searchEntity){
		return $http.post("../brandShenHe/search.do?page="+page+"&rows="+rows,searchEntity);
	}


	
	this.selectOptionList = function(){
		return $http.get("../brand/selectOptionList.do");
	}

    this.updateStatus = function(ids,status){
        return $http.get('../brandShenHe/updateStatus.do?ids='+ids+"&status="+status);
    }
});